File: frontend/src/types/artifact.ts
export type ArtifactType =
  | "high_level_design"
  | "detailed_design"
  | "code_summary"
  | "test_summary"
  | "generated_file";

export type Artifact = {
  id: string;
  run_id: string;
  project_id: string;
  artifact_type: ArtifactType;
  title: string;
  content: string;
  created_at: string;
  metadata?: Record<string, unknown>;
};

File: backend/app/models/artifact.py
from datetime import datetime
from typing import Literal

from pydantic import BaseModel

ArtifactType = Literal[
    "high_level_design",
    "detailed_design",
    "code_summary",
    "test_summary",
]


class Artifact(BaseModel):
    id: str
    run_id: str
    project_id: str
    artifact_type: ArtifactType
    title: str
    content: str
    created_at: datetime
